### PR TITLE
Cleanup naming in filebeat for better readability

### PR DESCRIPTION
**Pull request by ruflin**
Cleanup naming of channel and loggers in filebeat to make it easier to understand what actually happens but keep the abstraction.
Channels and loggers with a specific tasks were moved to the relevant packages

### CODE REVIEW

File: filebeat/beater/filebeat.go [hunk 1]
```diff
@@ -82,19 +81,27 @@ func (fb *Filebeat) Run(b *beat.Beat) error {
 		logp.Err("Could not start registrar: %v", err)
 	}
 	// Stopping registrar will write last state
-	defer registrar.Stop()
+	defer func() {
```
Comment (ruflin) [inline filebeat/beater/filebeat.go]: @urso can you have a really close look at the following defer statements which I have changed? I moved the `registrarLogger.Close()`which was `sucessLogger `before, before the registrar Stopping as I think that makes more sense. Same change I did for the publisherChan below.

Comment (urso) [inline filebeat/beater/filebeat.go]: This change here should not matter much, due to registrar normally not being blocked for long times. But `registrarLogger` using a channel + if registrar would block, publisher would be blocked as well.

File: filebeat/beater/filebeat.go [hunk 2]
```diff
@@ -82,19 +81,27 @@ func (fb *Filebeat) Run(b *beat.Beat) error {
...
 	publisher.Start()
 	// Stopping publisher (might potentially drop items)
-	defer publisher.Stop()
-	defer successLogger.Close()
+	defer func() {
+		// Closing channel to publisher
+		publisherChan.Close()
+		// Stopping publisher itself
+		publisher.Stop()
+	}()
```
Comment (urso) [inline filebeat/beater/filebeat.go]: closing `publisherChan` after spooler can make the spooler hang if:
- publisher is blocked by output
- channel being full
- spooler is trying to push via publisherChan

=> shutdown hangs

File: filebeat/beater/filebeat.go (hunk 1 above)
Comment (ruflin) [inline filebeat/beater/filebeat.go]: I reverted this change and add a comment.

**Comment by ruflin** [comment]
Closing in favor of https://github.com/elastic/beats/pull/2584

**Pull request closed without merging**